prowler/providers/aws/services/organizations/organizations_scp_check_deny_regions/organizations_scp_check_deny_regions.py

```diff
@@ -0,0 +1,102 @@
+                            # Deny if Condition = {"StringNotEquals": {"aws:RequestedRegion": [region1, region2]}}
+                            if (
+                                statement.get("Effect") == "Deny"
+                                and "Condition" in statement
+                                and "StringNotEquals" in statement["Condition"]
+                                and "aws:RequestedRegion"
+                                in statement["Condition"]["StringNotEquals"]
+                            ):
+                                if (
+                                    organizations_enabled_regions
+                                    == statement["Condition"]["StringNotEquals"][
+                                        "aws:RequestedRegion"
+                                    ]
+                                ):
+                                    # We found the statement, no need to continue
+                                    scp_check_deny_all_regions = True
+                                    break
+                                else:
+                                    # We found a statement restricting some regions, we continue to check if there is another one more restrictive
+                                    scp_check_deny_some_regions = True
+
+                            # Allow if Condition = {"StringEquals": {"aws:RequestedRegion": [region1, region2]}}
+                            if (
+                                policy.content.get("Statement") == "Allow"
+                                and "Condition" in statement
+                                and "StringEquals" in statement["Condition"]
+                                and "aws:RequestedRegion"
+                                in statement["Condition"]["StringEquals"]
+                            ):
+                                if (
+                                    organizations_enabled_regions
+                                    == statement["Condition"]["StringEquals"][
+                                        "aws:RequestedRegion"
+                                    ]
+                                ):
+                                    # We found the statement, no need to continue
+                                    scp_check_deny_all_regions = True
+                                    break
+                                else:
+                                    # We found a statement restricting some regions, we continue to check if there is another one more restrictive
+                                    scp_check_deny_some_regions = True
+
+                        if scp_check_deny_all_regions:
+                            report.status = "PASS"
+                            report.status_extended = (
+                                f"SCP policy restricting regions found: {policy.id}"
+                            )
+                            findings.append(report)
```

Comment:
I think you can call this once at the end of the function instead of having some right?

Reply:
i improved this. thanks.